flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/reader/deserializer/MongoJsonDeserializationSchema.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.mongodb.source.reader.deserializer;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+
+import org.bson.BsonDocument;
+import org.bson.json.JsonMode;
+
+import java.util.Optional;
+
+/**
+ * A schema bridge for deserializing the MongoDB's {@code BsonDocument} to MongoDB's {@link
+ * JsonMode}'s RELAXED Json string.
+ */
+@PublicEvolving
+public class MongoJsonDeserializationSchema implements MongoDeserializationSchema<String> {
```

Comment:
What's the use-case for this schema?

Reply:
Used to convert `BsonDocument` to [Json format](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/).

```java
    public void test() throws Exception {
        String database = "test";
        String collection = "test_coll";

        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        MongoSource<String> mongoSource =
                MongoSource.<String>builder()
                        .setUri("mongodb://127.0.0.1:27017")
                        .setDatabase("test")
                        .setCollection("test")
                        .setDeserializationSchema(new MongoJsonDeserializationSchema())
                        .build();

        env.fromSource(mongoSource, WatermarkStrategy.noWatermarks(), "MongoSource")
                .setParallelism(1)
                .print();

        env.execute();
    }
```

Output Example
```json
{"_id": {"$oid": "63803fa3fdca9606df01da9e"}, "f0": 1, "f1": "hRKnchMIFNZMfAtBtYRViNleBezFgTSq"}
{"_id": {"$oid": "63803fa3fdca9606df01da9f"}, "f0": 2, "f1": "oLRoMyLXvUxFwmOUINDwwhJeKpaSDJiM"}
{"_id": {"$oid": "63803fa3fdca9606df01daa0"}, "f0": 3, "f1": "esOnWaOwCUcsEGqisWNrupkbmIKtPZVS"}
{"_id": {"$oid": "63803fa3fdca9606df01daa1"}, "f0": 4, "f1": "lEqQYSNJKjUNILHFDnFQugpijBMXbqpy"}
{"_id": {"$oid": "63803fa3fdca9606df01daa2"}, "f0": 5, "f1": "lcgroaWbztgNIxUHwbVoVmSSLpmBYvXC"}
{"_id": {"$oid": "63803fa3fdca9606df01daa3"}, "f0": 6, "f1": "ELFXtzuJpLkWuIlXRVChRBqxnklbPGQF"}
{"_id": {"$oid": "63803fa3fdca9606df01daa4"}, "f0": 7, "f1": "WtFiNqcHCYmriyYFLBZtyqMPQDEidULJ"}
{"_id": {"$oid": "63803fa3fdca9606df01daa5"}, "f0": 8, "f1": "VDmoZYzuANQfpQtTnUdwXueNILPUgyNq"}
{"_id": {"$oid": "63803fa3fdca9606df01daa6"}, "f0": 9, "f1": "gLSwHUyzTHJjyOBrkjKlWTfsjtMeedef"}
{"_id": {"$oid": "63803fa3fdca9606df01daa7"}, "f0": 10, "f1": "hzHIcuDdvnEHcQYsKmPkbtmmzhWIyOTu"}
```

Reply:
Moved it into MongoSourceITCase.